ui/src/app/base/components/NotificationGroup/NotificationGroup.test.tsx

```diff
@@ -143,55 +118,39 @@ describe("NotificationGroup", () => {
 
   it("can dismiss a single notification", () => {
     const store = mockStore(state);
-    const notifications = [
-      {
-        id: 1,
-        category: "error",
-        message: "an error occurred",
-        dismissable: true,
-      },
-    ];
+    const notifications = [notificationFactory({ dismissable: true })];
 
     const wrapper = mount(
       <Provider store={store}>
         <NotificationGroup notifications={notifications} type="negative" />
       </Provider>
     );
 
-    wrapper.find("button[data-test='action-link']").simulate("click");
+    wrapper.find("button.p-icon--close").simulate("click");
```

Comment:
Seems a shame to revert using `data-test` here, was that unavoidable? Best to avoid using classes for selectors if possible.

Reply:
Yeah that's a part of react-components. I think using classes is going to be unavoidable in a lot of cases.